MetasysServices/MetasysClient.cs

```diff
@@ -1207,7 +1209,58 @@ private async Task SendCommandRequestAsync(Guid id, string command, IEnumerable<
             }
         }
 
+        #region "ad-hoc calls" // =========================================================================================================
+        // Support Ad-Hoc calls -----------------------------------------------------------------------------------------------------------
+        ///<inheritdoc/>
+        public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage requestMessage, HttpCompletionOption completionOption = HttpCompletionOption.ResponseContentRead, CancellationToken cancellationToken = default)
+        {
+            var response = new HttpResponseMessage();
+            try
+            {
+                var baseUri = new Uri(Client.BaseUrl);
+
+                if (!baseUri.IsBaseOf(requestMessage.RequestUri))
+                {
+                    //By default it failed with 401 unauthorized. Do we need to handle it explicitly
+                    //Requesting to connect different host.
+                }
+
+                var flurlRequest = Client.Request();
+                flurlRequest.Url = GetUrlFromFromRequest(baseUri, requestMessage);
+
+                //Copy Headers
+                foreach (var header in requestMessage.Headers)
+                {
+                    flurlRequest.Headers.Add(header.Key, header.Value);
+                }
+
+                response = await flurlRequest.SendAsync(requestMessage.Method, requestMessage.Content, cancellationToken, completionOption).ConfigureAwait(false);
+            }
+            catch (FlurlHttpException e)
+            {
+                ThrowHttpException(e);
+            }
+            return response;
+        }
+
+        private string GetUrlFromFromRequest(Uri baseUri, HttpRequestMessage requestMessage)
+        {
+            UriBuilder uriBuilder = new UriBuilder(baseUri.AbsoluteUri);
+
+            if (requestMessage.RequestUri.IsAbsoluteUri)
+            {
+                uriBuilder.Path = requestMessage.RequestUri.LocalPath;
+                uriBuilder.Query = requestMessage.RequestUri.Query;
+                uriBuilder.Fragment = requestMessage.RequestUri.Fragment;
+            }
+            else
+            {
+                uriBuilder.Path = requestMessage.RequestUri.ToString();
+            }
 
+            return uriBuilder.ToString();
+        }

```

Comment:
I'm not understanding exactly what this method is trying to achieve.

If the request uses an absolute URI, then you are ignoring the scheme, server, and port from `requestMessage`?

But if it's a relative URI, then you are ignoring the query and fragment?

Reply:
`uriBuilder` is created with the `baseUri.AbsoluteUri` so it will take the schem, host and port information from the `baseUri`.
If the request uses absolute Uri then will copy the Path, Query and Fragment because I assume that will support the calls to the host with which the `MetasysClient` instance is created.

If the request uses relative Uri then there will be no Path, Query and Fragment in it so I skipped it.

Reply:
> If the request uses relative Uri then there will be no Path, Query and Fragment in it so I skipped it.

What makes you say that?

A relative URI could certainly be like `"/equipment?pageSize=5"`, which has a path and query parameter.

I see you made changes to this function, so this thread may no longer be relevant.